objectivec/include/ort_checkpoint.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+#include <stdint.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * An ORT checkpoint is a snapshot of the state of a model at a given point in time.
+ *
+ * This class holds the entire training session state that includes model parameters,
+ * their gradients, optimizer parameters, and user properties. The ORTTrainingSession leverages the
+ * ORTCheckpointState by accessing and updating the contained training state.
+ *
+ * Available since v1.16.0.
+ *
+ * @note Note that the training session created with a checkpoint state uses this state to store the entire training
+ * state (including model parameters, its gradients, the optimizer states and the properties). The ORTTraingSession
+ * does not hold a copy of the checkpoint state. Therefore, it is required that the checkpoint state outlive the
+ * lifetime of the training session.
+ */
+@interface ORTCheckpoint : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Creates a checkpoint from directory on disk.
+ *
+ * @param path The path to the checkpoint directory.
+ * @param error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithPath:(NSString*)path
```

Comment:
IIRC, the checkpoint construction requires the default logger to be initialized, right? if so, maybe we should either pass in the ORTEnv as a parameter here or document this requirement.

Reply:
@baijumeswani From training point of view, should we go with including ORTEnv as a parameter in Objective C API or just document the requirement